darwin/item.py

```diff
@@ -19,10 +59,101 @@ class DatasetItem:
 
     @property
     def full_path(self) -> str:
+        """
+        The full POSIX relative path of this ``DatasetItem``.
+        """
         return construct_full_path(self.path, self.filename)
 
+    @classmethod
+    def parse(cls, raw: Dict[str, Any]) -> "DatasetItem":
+        """
+        Parses the given dictionary into a ``DatasetItem``. Raises if such is not possible.
+
+        Parameters
+        ----------
+        raw : Dict[str, Any]
+            The dictionary to parse.
+
+        Returns
+        -------
+        DatasetItem
+            A dataset item with the parsed information.
+
+        Raises
+        ------
+        ValueError
+            If any of the keys from the given dictionary do not have the correct format.
+        """
+        id: int = raw["id"]
+        if not isinstance(id, int):
+            raise ValueError("Key 'id' must have an integer for a value.")
+
+        filename: str = raw["filename"]
+        if not isinstance(filename, str):
+            raise ValueError("Key 'filename' must have a string for a value.")
+
+        status: str = raw["status"]
+        if not isinstance(status, str):
+            raise ValueError("Key 'status' must have a string for a value.")
+
+        archived: bool = raw["archived"]
+        if not isinstance(archived, bool):
+            raise ValueError("Key 'archived' must have a boolean for a value.")
 
+        file_size: int = raw["file_size"]
+        if not isinstance(file_size, int):
+            raise ValueError("Key 'file_size' must have an integer for a value.")
+
+        dataset_id: int = raw["dataset_id"]
+        if not isinstance(dataset_id, int):
+            raise ValueError("Key 'dataset_id' must have an integer for a value.")
+
+        sequence: int = raw["seq"]
+        if not isinstance(sequence, int):
+            raise ValueError("Key 'seq' must have an integer for a value.")
+
+        workflow_id: Optional[int] = raw.get("current_workflow_id")
+        if not isinstance(workflow_id, int) and workflow_id is not None:
+            raise ValueError("Key 'workflow_id' must have an integer for a value or be nil.")
+
+        path: str = raw["path"]
+        if not isinstance(path, str):
+            raise ValueError("Key 'path' must have a string for a value.")
+
+        return DatasetItem(
+            id,
+            filename,
+            status,
+            archived,
+            file_size,
+            dataset_id,
+            "n/a",
+            sequence,
+            workflow_id,
+            path,
+        )
```

Comment:
I personally don't like this `if not` validation logic for each item in the dictionary. Feels off from best practices tbh. Would something like this still work?

```suggestion
    @classmethod
    def parse(cls, **raw: Dict[str, Any]) -> "DatasetItem":
        """
        Parses the given dictionary into a ``DatasetItem``. Raises if such is not possible.

        Parameters
        ----------
        raw : Dict[str, Any]
            The dictionary to parse.

        Returns
        -------
        DatasetItem
            A dataset item with the parsed information.

        Raises
        ------
        ValueError
            If any of the keys from the given dictionary do not have the correct format.
        """

        return DatasetItem(**kwargs)
```

Reply:
I have never used kargs.. I will look into it.
I am also not  a fan of this form of validation, but the only other option I know is JSonSchema, which I think is totally overkill for this.

Reply:
Doing it this way would not:
 - perform validation on the Parse function, which is something we should do not only because we do it in other parse functions but also because we are converting a potentially unwanted dictionary into an object we understand
 - force `datase_slug` to be "n/a", which is the case with the original function

I am afraid using the unpacking operator is not going to work for this case.